src/coreclr/jit/lower.cpp

```diff
@@ -6393,12 +6393,19 @@ PhaseStatus Lowering::DoPhase()
     {
         comp->optLoopsMarked = false;
         bool modified        = comp->fgUpdateFlowGraph();
+        modified |= comp->fgRemoveDeadBlocks();
```

Comment:
If `modified` returned false, do we need to run this?

Reply:
Yes. This should be run regardless (as we do today). I want to do it after update flow graph because it might make more blocks eligible for removing.

Reply:
Makes sense.